src/models/Capability.js

```diff
@@ -2,15 +2,25 @@ import DB from "../lib/DB";
 import InsufficientPermissionsError from "../errors/InsufficientPermissionsError";
 
 export default class Capability {
-  static async getCapabilitiesByLoggedInUser(session_token) {
+  static async getCapabilitiesByLoggedInUser(session_token, params) {
     const db = new DB();
 
-    const sql = `SELECT DISTINCT capabilities.name from people_groups 
+    let sql = `SELECT DISTINCT capabilities.name from people_groups 
     LEFT JOIN groups ON groups.id = people_groups.group_id
     LEFT JOIN group_capabilities ON group_capabilities.group_id = groups.id
     LEFT JOIN capabilities ON capabilities.id = group_capabilities.capabilities_id
     WHERE people_groups.people_id IN (select people_id from sessions where session = :session_token) AND capabilities.name IS NOT NULL`;
 
+    if (params.length > 0) {
+        for (let i = 0; i < params.length; i++) {
+            if(params[i]) {
+                sql += ` AND params${i + 1} = ${params[i]}`;
+            } else {
+                sql += ` AND params${i + 1} IS NULL`;
+            }
+        }
+    }
+
```

Comment:
I don't think we need the loop here. We can do the following:
1.) Instead of having `...params`, just provide param1, param2, and param3. THey default to null anyway if the caller did not provide them.

2.) If param1 == null, then query `PARAM1 IS NULL`. We don't need to make the params conditional. We can include the three PARAMS all the time in  the query.